src/Analyzer/FunctionNode.h

```diff
@@ -124,24 +133,18 @@ class FunctionNode final : public IQueryTreeNode
     /** Get non aggregate function.
       * If function is not resolved nullptr returned.
       */
-    const FunctionOverloadResolverPtr & getFunction() const
-    {
-        return function;
-    }
+    FunctionBasePtr getFunction() const;
```

Comment:
btw is `FunctionBasePtr` needed here? Can we return just `const IFunctionBase*` ?

Reply:
Yes, it is used during the planning step to build action dag.